frontend/src/Calendar.js

```diff
@@ -2,19 +2,108 @@ import React from 'react'
 import FullCalendar from '@fullcalendar/react'
 import dayGridPlugin from '@fullcalendar/daygrid'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
+// the conversion of one minute to 60000 milliseconds
+const MIN_TO_MS = 60000;
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
+      schedulesTimes: [],
+      scheduleIdx: 0     // Current schedule
+    }
+  }
+
+  componentDidMount(){
+    this.getAllSchedules()
+  }
+
+  async getAllSchedules() {
+    const schedules = await fetchUserCalendar()
+    for(const schedule of schedules) {
+      this.getMeetingTimes(schedule)
+    }
+    
+    this.setState({
+      fetchDone: true
+    })
+  }
+
+  getNextSchedule() {
+    if(this.state.scheduleIdx < this.state.schedulesTimes.length - 1) {
+      this.setState({scheduleIdx : this.state.scheduleIdx + 1})
+    }
+  }
+
+  getPrevSchedule() {
+    if(this.state.scheduleIdx > 0) {
+      this.setState({scheduleIdx : this.state.scheduleIdx - 1})
+    }
+  }
+
+  getMeetingTimes(schedule) {
```

Comment:
This function name doesn't reflect what the function does

Reply:
Changed name to `generateScheduleMeetingTimes() `